### PR TITLE
Refactor views

### PR DESCRIPTION
**Pull request by jaredhirsch**
I noticed a lot of near-duplicate code in the views, where we had copy-pasted functions that only varied on the name of the model property.

To eliminate the duplication, I gave the models a uniform API, then extracted the common view methods to a row item view superclass. Seems nicer, except that I've gone with `result` and `resultType` to capture the url/suggestion variation; suggestions for alternative names are welcome :-)

This PR is probably easier to read one commit at a time, rather than skimming the complete diff.

@chuckharmston r?

### CODE REVIEW

File: src/views/search_suggestions/search_suggestions_item_view.js [hunk 1]
```diff
@@ -1,24 +1,9 @@
...
 
-export default BaseView.extend({
+export default RowItemView.extend({
   template: SearchSuggestionsItemTemplate,
 
-  events: {
-    'mousedown': 'openSuggestion',
-    'select': 'sendSelectionDetails'
-  },
+  events: RowItemView.prototype.events
```
Comment (chuckharmston) [inline src/views/search_suggestions/search_suggestions_item_view.js:7]: If we're `extend`ing `RowItemView` here, do we even need this?

Comment (chuckharmston) [inline src/views/search_suggestions/search_suggestions_item_view.js:7]: /g

File: src/models/activity_result.js [hunk 2]
```diff
@@ -3,6 +3,18 @@ import State from 'ampersand-state';
 export default State.extend({
   extraProperties: 'allow',
 
+  resultType: 'url',
+  derived: {
```
Comment (chuckharmston) [inline src/models/activity_result.js:7]: If we're adding `derived` (which I just learned existed; neat!), could you move `displayTitle` and `displayImage` in there too?

File: src/models/recommended_result.js [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+    result: {
+      deps: ['url'],
+      fn: function() {
+        return this.url;
+      },
+      cache: false
+    }
+  },
+
   faviconUrl () {
```
Comment (chuckharmston) [inline src/models/recommended_result.js]: This into `derived` as well?

**Comment by chuckharmston** [comment]
This looks great, nicely done. Feel free to merge it in whenever you work out whether my thoughts were of any value.

File: src/views/search_suggestions/search_suggestions_item_view.js (hunk 1 above)
Comment (jaredhirsch) [inline src/views/search_suggestions/search_suggestions_item_view.js:7]: Sadly, `events` aren't inherited on views, so this is the awkward recommended approach. Have a look at the paragraph just above this point in the docs: https://github.com/AmpersandJS/ampersand-view#bindings

File: src/models/activity_result.js (hunk 2 above)
Comment (jaredhirsch) [inline src/models/activity_result.js:7]: Yes, great idea

File: src/models/recommended_result.js (hunk 3 above)
Comment (jaredhirsch) [inline src/models/recommended_result.js]: :+1:

**Pull request merged**